service/models.py

```diff
@@ -1,227 +1,219 @@
 # Create the SQLAlchemy object to be initialized later in init_db()
 db = SQLAlchemy()
 
 
-# Function to initialize the database
+# Function to initialize the tables in DB
 def init_db(app):
-    """ Initializes the SQLAlchemy app """
+    """ Initialize the Shopcart and the Item tables through SQLAlchemy """
     Shopcart.init_db(app)
     Item.init_db(app)
 
 
 class DataValidationError(Exception):
-    """ Used for an data validation errors when deserializing """
+    """ Used for object deserialization data validation errors """
 
 
+# TODO: create a shared base class for Shopcart and Item
 class Shopcart(db.Model):
-    """
-    Class that represents a Shopcart
-    """
+    """ The Shopcart Table """
 
     app = None
 
     # Table Schema
-    # customer_id (primary key), customer_name, shopcart_quantity, shopcart_price
-    customer_id = db.Column(db.Integer, primary_key=True)
-    customer_name = db.Column(db.String(63), nullable=False)
+    id = db.Column(db.Integer, primary_key=True)    # correspond to customer_id
+    name = db.Column(db.String(63), nullable=False) # correspond to customer_name
+    items = db.relationship("Item", backref="shopcart", passive_deletes=True)
 
     def __repr__(self):
-        return f"<Shopcart for customer {self.customer_name} id=[{self.customer_id}]>"
+        return f"{type(self).__name__}({self.id}, {self.name})"
 
-    def create(cls, self):
-        """
-        Creates a Shopcart to the database
-        """
-        logger.info("Creating %s %s", cls.__name__, self.__repr__)
-        self.customer_id = None  # pylint: disable=invalid-name
+    def create(self):
+        """ Create a shopcart in DB table """
+        logger.info("Create %s", self.__repr__)
+        self.id = None  # pylint: disable=invalid-name
         db.session.add(self)
         db.session.commit()
 
-    def update(cls, self):
-        """
-        Updates a Shopcart to the database
-        """
-        logger.info("Saving %s %s", cls.__name__, self.__repr__)
+    def update(self):
+        """ Update a shopcart in DB table """
+        logger.info("Update %s", self.__repr__)
         db.session.commit()
 
-    def delete(cls, self):
-        """ Removes a Shopcart from the data store """
-        logger.info("Deleting %s %s", cls.__name__, self.__repr__)
+    def delete(self):
+        """ Delete a shopcart in DB table """
+        logger.info("Delete %s", self.__repr__)
         db.session.delete(self)
         db.session.commit()
 
     def serialize(self):
-        """ Serializes a Shopcart into a dictionary """
-        return {
-            "customer_id": self.customer_id, 
-            "customer_name": self.customer_name,
+        """ Serialize a shopcart object into a dictionary """
+        shopcart = {
+            "id": self.id,
+            "name": self.name,
+            "items": list()
         }
+        for item in self.items:
+            shopcart["items"].append(item.serialize())
+        return shopcart
```

Comment:
Almost the same in the serialize function but I wrote **"items": []** at line 75

Reply:
ohhh, I'm good with both! Feel free to change it to `"items": []` 🙌🏻